src/App.js

```diff
@@ -1,24 +1,216 @@
+                  <small className="text-muted">
+                    {dayjs(activity.startTime).format("ddd[,] D MMM [at] H:mm")} •{" "}
+                    {activity.instructor}
+                  </small>
+                </li>
+              );
+            })}
+          </ol>
+          <hr />
+        </div>
+      );
+    }
+  }
+
+  function nextWeeksActivitiesComponent() {
+    if (nextWeeksActivityList.length !== 0) {
+      return (
+        <div>
+          <h3>Next Week</h3>
+          <ol className="list-unstyled">
+            {nextWeeksActivityList.map((activity) => {
+              return (
+                <li className="hover-light-bg p-2 rounded-2" key={activity.id}>
+                  <header>{activity.title}</header>
+                  <small className="text-muted">
+                    {dayjs(activity.startTime).format(
+                      "ddd[,] DD MMM [at] H:mm"
+                    )}{" "}
+                    • {activity.instructor}
+                    {}
+                  </small>
+                </li>
+              );
+            })}
+          </ol>
+          <hr />
+        </div>
+      );
+    }
+  }
+
+  function futureActivitiesComponent() {
+    if (futureActivityList.length !== 0) {
+      return (
+        <div>
+          <h3>Future</h3>
+          <ol className="list-unstyled">
+            {futureActivityList.map((activity) => {
+              return (
+                <li className="hover-light-bg p-2 rounded-2" key={activity.id}>
+                  <header>{activity.title}</header>
+                  <small className="text-muted">
+                    {dayjs(activity.startTime).format("D MMM [at] H:mm")} •{" "}
+                    {activity.instructor}
+                  </small>
+                </li>
+              );
+            })}
+          </ol>
+          <hr />
+        </div>
+      );
+    }
+  }
+
+  useEffect(() => {
+    todaysActivities();
+    tomorrowsActivities();
+    restOfWeekActivities();
+    nextWeeksActivities();
+    futureActivities();
+  }, []);
+
   return (
     <div className="container p-3">
       <h1 className="mb-5">Schedule</h1>
       <div className="row">
         <div className="col-lg-6 col-md-8">
-          <h2 className="fs-6">Showing all</h2>
+          <h3 className="fs-6">Showing all</h3>
 
-          <ol className="list-unstyled">
-            {activities.map((activity) => (
-              <li className="hover-light-bg p-2 rounded-2" key={activity.id}>
-                <header>{activity.title}</header>
-                <small className="text-muted">
-                  {dayjs(activity.startTime).format("D MMM [at] H:mm")} • {activity.instructor}
-                </small>
-              </li>
-            ))}
-          </ol>
+          {todaysActivitiesComponent()}
+          {tomorrowsActivitiesComponent()}
+          {restOfWeekActivitiesComponent()}
+          {nextWeeksActivitiesComponent()}
+          {futureActivitiesComponent()}
```

Comment:
It's better practice to extract this into a separate component. Instead of using functions.
You'll also realize you can refactor them into a single component and pass different props for each section.

> You can perform these refactors with the time left.